mlflow/tracking/client.py

```diff
@@ -497,10 +501,35 @@ def create_model_version(self, name, source, run_id, tags=None):
         :param run_id: Run ID from MLflow tracking server that generated the model
         :param tags: A dictionary of key-value pairs that are converted into
                      :py:class:`mlflow.entities.model_registry.ModelVersionTag` objects.
+        :param run_link: Link to the run from an MLflow tracking server that generated this model.
         :return: Single :py:class:`mlflow.entities.model_registry.ModelVersion` object created by
                  backend.
         """
-        return self._get_registry_client().create_model_version(name, source, run_id, tags)
+        tracking_uri = self._tracking_client.tracking_uri
+        # for Databricks backends, we support automatically populating the run link field
+        if is_databricks_uri(tracking_uri) and tracking_uri != self._registry_uri and not run_link:
+            # if using the default Databricks tracking URI and in a notebook, we can automatically
+            # figure out the run-link.
+            if is_databricks_default_tracking_uri(tracking_uri) and is_in_databricks_notebook():
+                # use DBUtils to determine workspace information.
+                workspace_host, workspace_id = get_workspace_info_from_dbutils()
+            else:
+                # in this scenario, we're not able to automatically extract the workspace ID
+                # to proceed, and users will need to pass in a databricks profile with the scheme:
+                # databricks://scope/prefix and store the host and workspace-ID as a secret in the
+                # Databricks Secret Manager with scope=<scope> and key=<prefix>-workspaceid.
+                workspace_host, workspace_id = \
+                    get_workspace_info_from_databricks_secrets(tracking_uri)
+            # retrieve experiment ID of the run for the URL
+            experiment_id = self.get_run(run_id).info.experiment_id
+            if workspace_host and run_id and experiment_id:
+                run_link = construct_run_url(workspace_host, experiment_id, run_id, workspace_id)
```

Comment:
I wonder if we should fail here if workspace_id is not given, or just give a warning. I guess throwing is okay - user can give a run_link explicitly to bypass this problem. I am not sure what the best way to document this is though - probably in the databricks docs for multi-workspace workflows.

Reply:
Yeah my concern was that ST users might not know what the workspace-id is, and so they might just not provide one in the secrets, so then that would just return None. In that scenario, it'd still be ok since it isn't needed in the URL anyways.

For MT users, if they didn't specify a workspace ID, would going to the URL just make them log in?

Reply:
mm yeah.. that's a good point. i think if they only have access to 1 workspace it will go to that one, and if not, put them in the workspace selector menu. maybe we can print out a message like "No workspace ID specified; if your Databricks workspaces share the same host URL, you may want to specify the workspace ID (along with the host information in the secret manager) for run lineage tracking." if no workspace_id is found.

Reply:
Yeah makes sense - will add that